src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryWriter.cs

```diff
@@ -778,16 +782,23 @@ public override unsafe void WriteDoubleText(double d)
             {
                 int offset;
                 byte[] buffer = GetTextNodeBuffer(1 + sizeof(double), out offset);
-                byte* bytes = (byte*)&d;
+                long value = *(long*)&d;
                 buffer[offset + 0] = (byte)XmlBinaryNodeType.DoubleText;
-                buffer[offset + 1] = bytes[0];
-                buffer[offset + 2] = bytes[1];
-                buffer[offset + 3] = bytes[2];
-                buffer[offset + 4] = bytes[3];
-                buffer[offset + 5] = bytes[4];
-                buffer[offset + 6] = bytes[5];
-                buffer[offset + 7] = bytes[6];
-                buffer[offset + 8] = bytes[7];
+                buffer[offset + 0] = (byte)value;
```

Comment:
This should just use `BinaryPrimitives.WriteDoubleLittleEndian`

Reply:
Agreed, that's much simpler.

Reply:
FWIW, there are more places that can use `BinaryPrimitives`, for example `WriteInt64` method.

Reply:
Do you want me to make that change as part of this PR?   I was hoping that this could get into net7 so fix CI there, so I was trying to keep the changes as small as possible.